openghg/store/_boundary_conditions.py

```diff
@@ -160,66 +184,39 @@ def read_file(
-            if "time" in bc_data.coords:
-                bc_data = update_zero_dim(bc_data, dim="time")
+        # Call appropriate standardisation function with input parameters
+        boundary_condition_data = parser_fn(**parser_input_parameters)
+        results: resultsType = defaultdict(dict)
 
+        for key, value in boundary_condition_data.items():
             # Currently ACRG boundary conditions are split by month or year
+            bc_data = value["data"]
             bc_time = bc_data["time"]
 
             start_date, end_date, period_str = infer_date_range(
                 bc_time, filepath=filepath, period=period, continuous=continuous
             )
 
             # Checking against expected format for boundary conditions
-            BoundaryConditions.validate_data(bc_data)
+            BoundaryConditions.validate_data(value["data"])
             data_type = "boundary_conditions"
 
-            metadata["start_date"] = str(start_date)
-            metadata["end_date"] = str(end_date)
-            metadata["data_type"] = data_type
-
-            metadata["max_longitude"] = round(float(bc_data["lon"].max()), 5)
-            metadata["min_longitude"] = round(float(bc_data["lon"].min()), 5)
-            metadata["max_latitude"] = round(float(bc_data["lat"].max()), 5)
-            metadata["min_latitude"] = round(float(bc_data["lat"].min()), 5)
-            metadata["min_height"] = round(float(bc_data["height"].min()), 5)
-            metadata["max_height"] = round(float(bc_data["height"].max()), 5)
+            additional_metadata["start_date"] = str(start_date)
+            additional_metadata["end_date"] = str(end_date)
+            additional_metadata["data_type"] = data_type
 
-            metadata["input_filename"] = filepath.name
+            additional_metadata["max_longitude"] = round(float(bc_data["lon"].max()), 5)
+            additional_metadata["min_longitude"] = round(float(bc_data["lon"].min()), 5)
+            additional_metadata["max_latitude"] = round(float(bc_data["lat"].max()), 5)
+            additional_metadata["min_latitude"] = round(float(bc_data["lat"].min()), 5)
+            additional_metadata["min_height"] = round(float(bc_data["height"].min()), 5)
+            additional_metadata["max_height"] = round(float(bc_data["height"].max()), 5)
 
-            metadata["time_period"] = period_str
+            additional_metadata["input_filename"] = filepath.name
 
-            key = "_".join((species, bc_input, domain))
+            additional_metadata["time_period"] = period_str
 
-            boundary_conditions_data: dict[str, dict] = {}
-            boundary_conditions_data[key] = {}
-            boundary_conditions_data[key]["data"] = bc_data
-            boundary_conditions_data[key]["metadata"] = metadata
-
-            matched_keys = set(metadata) & set(fn_input_parameters)
+            matched_keys = set(bc_data) & set(fn_input_parameters)
```

Comment:
`bc_data` is a dataset, so it acts like a mapping from data variable names to data arrays, so `set(bc_data)` will be the data variable names from `bc_data`. In the old code, it looks like it should be the set of keys in the metadata though.

Reply:
I actually don't think we need this or the separate `additional_input_parameters` dictionary defined below anymore if we've moved to using the parse framework. The new L151 in the openghg/store/_boundary_conditions.py file creates the `additional_input_parameters` dictionary against the parse function itself.

I think the previous set up was a temporary stopgap before we moved to the parse framework for this since we didn't have the parse function call to compare against.